internal/runners/auth/signup.go

```diff
@@ -17,11 +19,14 @@ func NewSignup(prime primeable) *Signup {
 	return &Signup{prime.Output(), prime.Prompt(), prime.Config()}
 }
 
-func (s *Signup) Run() error {
-	err := authlet.Signup(s.Configurable, s.Outputer, s.Prompter)
-	if err != nil {
-		return err
+func (s *Signup) Run(interactive bool) error {
+	if authentication.LegacyGet().Authenticated() {
+		return locale.NewInputError("err_auth_authenticated", "You are already authenticated as: {{.V0}}. You can log out by running `state auth logout`.", authentication.LegacyGet().WhoAmI())
 	}
 
-	return nil
+	if !interactive {
+		return authlet.AuthenticateWithDevice(s.Outputer) // user can sign up from this page too
+	} else {
+		return authlet.Signup(s.Configurable, s.Outputer, s.Prompter)
+	}
```

Comment:
nit: since you're returning in the `if` case the else is unecessary
```suggestion
	if !interactive {
		return authlet.AuthenticateWithDevice(s.Outputer) // user can sign up from this page too
	}
	return authlet.Signup(s.Configurable, s.Outputer, s.Prompter)
```

Reply:
Okay